web_timeline/static/src/js/timeline_renderer.js

```diff
@@ -421,11 +421,16 @@ odoo.define('web_timeline.TimelineRenderer', function (require) {
             } else {
                 group = -1;
             }
-            _.each(self.colors, function (color) {
-                if (eval("'" + evt[color.field] + "' " + color.opt + " '" + color.value + "'")) {
-                    self.color = color.color;
-                }
-            });
+            var color = null;
+            if (self.arch.attrs.color_field !== undefined) {
+                color = evt[self.arch.attrs.color_field];
+            } else {
+                _.each(self.colors, function (col) {
+                    if (eval("'" + evt[col.field] + "' " + col.opt + " '" + col.value + "'")) {
+                        color = col.color;
```

Comment:
break in this case for not continuing evaluating, isn't it?

Reply:
@pedrobaeza 
I agree that we could add a break there to avoid evaluate the other colors. 
But,
- There is no 'break' for the _.each
- This is a part of the existing code (I apply the original code when no color_field is set)
- This loop is done on the timeline options specified on the view, so there will be only a few color to check. The time consumed will be insignificant
I'm not sure that it is necessary to change that knowing the little change that it will bring.
Let me know what do you think.

Reply:
Yeah, indeed this is improving existing code, but if no possible break here, let's keep it as is.